src/broker/broker.c

```diff
@@ -117,6 +117,10 @@ int broker_new(Broker **brokerp, const char *machine_id, int log_fd, int control
         if (r)
                 return error_fold(r);
 
+        r = sockopt_get_peerpidfd(controller_fd, broker->bus.pid, &broker->bus.pid_fd, &broker->bus.pid_fd_trusted);
+        if (r < 0 && r != -EOPNOTSUPP)
+                return error_trace(r);
```

Comment:
(We do this, because it is just too fragile to ensure that a given errno comes from the exact syscall the caller expects. In this case, you are fine beacuse you only call two syscalls, and both would return the expected EOPNOTSUPP.)